src/lm.jl

```diff
@@ -156,7 +156,30 @@ function coeftable(mm::LinearModel)
               ["x$i" for i = 1:size(mm.pp.X, 2)], 4)
 end
 
-predict(mm::LinearModel, newx::Matrix) =  newx * coef(mm)
+# Different methods of prediction for the different factorization types
+
+predict(mm::LinearModel, newx::AbstractMatrix) = newx * coef(mm)
+
+function predict(mm::LinearModel, newx::AbstractMatrix, interval_type::Symbol, level = 0.95)
+    prediction = newx * coef(mm)
+    interval_type == :confint || error("only :confint is currently implemented") #:predint will be implemented
+
+    R = cholfact!(mm.pp)[:U] #get the R matrix from the QR factorization
+    residvar = (ones(3,1) * deviance(mm)/dof_residual(mm))'
+
+    function _local(x::StridedVector)
+        Ac_ldiv_B!(R,x)
+        residvar * (x⋅x)
+    end
+
+    interval = similar(prediction)
+    for i in 1:length(interval)
+        interval[i] = _local(newx[i,:])[1]
+    end
+
+    interval = quantile(TDist(dof_residual(mm)), (1 - level)/2) * sqrt(interval)
```

Comment:
Right, now I tested essentially the version you suggest under 1 and the one that uses the dot product. I also for comparison tested the one I originally wrote (except that my `getR` function is implemented with `cholfact!` as suggested), which is half the number of lines, and in my mind much more intuitive (see below). 
I think I have spent enough time now playing around with the implementation. If you think it works now I would be willing to write up the tests, docstrings, prediction intervals and take a stab at the GLM code, when I find some time again.
```julia
function predict_suggested(mm::LinearModel, newx::AbstractMatrix, interval_type::Symbol, level = 0.95)

    interval_type == :confint || error("only :confint is currently implemented") #:predint will be implemented

    R = cholfact!(mm.pp)[:U] #get the R matrix from the QR factorization
    residvar = (ones(size(newx,2),1) * deviance(mm)/dof_residual(mm))'

    function _local(x::StridedVector)
        pred = x' * coef(mm)
        Ac_ldiv_B!(R,x)
        pred[1], (residvar * (x⋅x))[1]
    end

    n = size(newx, 1)
    retmean, retvariance = similar(newx, n), similar(newx, n)
    for i in 1:length(retmean)
        retmean[i], retvariance[i] = _local(newx[i,:])
    end

    interval = quantile(TDist(dof_residual(mm)), (1 - level)/2) * sqrt(retvariance)
    retmean, retmean .+ interval, retmean .- interval
end

function predict_suggested_with_dot(mm::LinearModel, newx::AbstractMatrix, interval_type::Symbol, level = 0.95)
    interval_type == :confint || error("only :confint is currently implemented") #:predint will be implemented

    R = cholfact!(mm.pp)[:U] #get the R matrix from the QR factorization
    residvar = (ones(size(newx,2),1) * deviance(mm)/dof_residual(mm))'

    function _local(x::StridedVector)
        pred = x ⋅ coef(mm)
        Ac_ldiv_B!(R,x)
        pred[1], (residvar * (x⋅x))[1]
    end

    n = size(newx, 1)
    retmean, retvariance = similar(newx, n), similar(newx, n)
    for i in 1:length(retmean)
        retmean[i], retvariance[i] = _local(newx[i,:])
    end

    interval = quantile(TDist(dof_residual(mm)), (1 - level)/2) * sqrt(retvariance)
    retmean, retmean .+ interval, retmean .- interval
end

function predict_original_PR(mm::LinearModel, newx::AbstractMatrix, interval_type::Symbol, level = 0.95)
    retmean = newx * coef(mm)
    interval_type == :confint || error("only :confint is currently implemented") #:predint will be implemented

    R = cholfact!(mm.pp)[:U] #get the R matrix from the QR factorization
    residvar = (ones(size(newx,2),1) * deviance(mm)/dof_residual(mm))
    tmp = (newx/R).^2
    retvariance = tmp * residvar #(newx/R).^2

    interval = quantile(TDist(dof_residual(mm)), (1 - level)/2) * sqrt(retvariance)
    retmean, retmean .+ interval, retmean .- interval
end
```
<img width="328" alt="skaermbillede 2016-12-16 kl 22 19 59" src="https://cloud.githubusercontent.com/assets/8429802/21279155/1f97dc36-c3df-11e6-86ac-8dee9770506d.png">